### DISCUSSION TITLE
Message Pointer Alignment

### DISCUSSION THREAD

**Discussion by carbon853** [uNetworking/uWebSockets#1551] [Q&A] [answered]
Hello!

I am serializing binary data across the network using `uWebSockets` and a serialization library.

In the message callback, the `std::string_view` is aligned according to a `MESSAGE_HEADER` offset. This offset differs from what the serialization library expects, and thus causes one of its internal checks to fail.

I can copy the data before deserializing to get an offset it likes, but I was wondering if there is an easier way to avoid this extra copy?

**Comment by uNetworkingAB** [comment, marked as answer]
There is no guarantee of alignment, not even to MESSAGE_HEADER.
